### DISCUSSION TITLE
Creating a counter

### DISCUSSION THREAD

**Discussion by kaeland** [rabbibotton/clog#71] [Q&A] [answered]
Hi Dr. Botton, 

I'm attempting to create a small counter app. So far, I've created two buttons for incrementing and decrementing a counter. From what I've seen in the tutorials, it looks like `connection-data-item` can/would be used to store the state of you application, in this case a count of some integer. In the code below, I've scripted the app to the best of my current understanding of CLOG. However, when I run the app in the browser, the count within the `p-element` doesn't update when I click the buttons. Do you have any advice for how I could approach this? 

-Kaeland 

```
(defpackage :todo-app-clog
  (:use :cl :clog))

(in-package :todo-app-clog)

(defun inc-count (obj)
  (print (connection-data-item obj "counter"))
  (incf (connection-data-item obj "counter")))

(defun dec-count (obj)
  (print (connection-data-item obj "counter"))
  (decf (connection-data-item obj "counter")))

(defun on-new-window (body)
  (setf (title (html-document body)) "Counter")
  (setf (connection-data-item body "counter") 0)
  (let* ((title (create-section body :h1 :content "Todo List:"))
         (inc-button (create-button body :content "Increase"))
         (dec-button (create-button body :content "Decrease"))
         (counter (create-section body :p :content (connection-data-item body "counter"))))
    (declare (ignorable title)
             (ignorable counter))
    (set-on-click inc-button #'inc-count)
    (set-on-click dec-button #'dec-count)
    (run body)))

(defun run-app ()
  (initialize #'on-new-window)
  (open-browser))
```

**Comment by kaeland** [comment]
@rabbibotton Just wanted to know your opinion on this? :D

**Comment by rabbibotton** [comment, marked as answer]
You never update the value of the counter:

    (set-on-click inc-button (lambda (obj)
                               ;; this works as we update the counter
                               (funcall #'inc-count obj)
                               (setf (text counter) (connection-data-item body "counter"))))
    ;; you are not updating the counter here
    (set-on-click dec-button #'dec-count)
